lib/Workflow.groovy

```diff
@@ -254,13 +254,36 @@ class Workflow {
         }
         if (total != (left + right)) {
             log.warn "=============================================================================\n" +
-                    "  Please check the name field (column 4) in the file supplied via --primer_bed.\n\n" +
-                    "  All of the values in that column do not end with those supplied by:\n" +
-                    "      --primer_left_suffix : $primer_left_suffix\n" +
-                    "      --primer_right_suffix: $primer_right_suffix\n\n" +
-                    "  This information is required to collapse the primer intervals into amplicons\n" + 
-                    "  for the coverage plots generated by the pipeline.\n" +
-                    "==================================================================================="
+                     "  Please check the name field (column 4) in the file supplied via --primer_bed.\n\n" +
+                     "  All of the values in that column do not end with those supplied by:\n" +
+                     "      --primer_left_suffix : $primer_left_suffix\n" +
+                     "      --primer_right_suffix: $primer_right_suffix\n\n" +
+                     "  This information is required to collapse the primer intervals into amplicons\n" + 
+                     "  for the coverage plots generated by the pipeline.\n" +
+                     "==================================================================================="
+        }
+    }
+
+    // Check if the primer BED file supplied to the pipeline is from the SWIFT/SNAP protocol
+    public static void checkIfSwiftProtocol(primer_bed_file, log) {
+        def count = 0
+        def line  = null
+        primer_bed_file.withReader { reader ->
+            while (line = reader.readLine()) {
+                def name = line.split('\t')[3]
+                if (name.contains('covid19genome')) {
```

Comment:
It might be that I am freaking out in advance, but should we provide 'covid19genome' as a variable to the function instead

Reply:
Yup! Another thing I thought I should do but chose not to! Man, your comments are like me with a massive case of deja vu! Love the reviews 🤩

Reply:
Hope this is good 😁